skimage/future/tests/test_graph.py

```diff
@@ -0,0 +1,13 @@
+import pytest
+
+
+def test_future_graph_import_error():
+    error_msg = "The API in skimage.future.graph was moved into skimage.graph"
+    with pytest.raises(ModuleNotFoundError, match=error_msg):
+        import skimage.future.graph
+
+    with pytest.raises(ModuleNotFoundError, match=error_msg):
+        from skimage.future import graph
```

Comment:
These two tests test the same import?

Reply:
Yes, the module and the contained `raise` statement is run regardless of the import statement. This even applies for the `from skimage.future.graph import x` case! You can see the test's passing in the CI for [the install from sdist job](https://github.com/scikit-image/scikit-image/actions/runs/3943584896/jobs/6748577634#step:5:271). Other jobs fail because they trigger the `raise` statement during doctest collection...

I haven't yet figured out the best approach to resolve this. We could likely use [`pytest_ignore_collect`](https://docs.pytest.org/en/latest/reference/reference.html#pytest.hookspec.pytest_ignore_collect) but I'd like to avoid that for a temporary patch.

Reply:
Using a warning instead would work. But the migration message doesn't appear as nicely at the end of the traceback..

Reply:
Interesting. Adding the following to our `conftest.py`

```python
def pytest_ignore_collect(collection_path, path, config):
    return str(collection_path).endswith("skimage/future/graph.py")
```

works to ignore the `ModuleNotFoundError` in `graph.py `. However, it then stumbles across https://github.com/scikit-image/scikit-image/blob/82331508becbca6a0735e4c1459eea2fb355b8ab/skimage/io/_plugins/gdal_plugin.py#L6-L8

I'm lost why that isn't the case if I remove `pytest_ignore_collect`. :thinking:

Reply:
Aha: https://github.com/scikit-image/scikit-image/blob/82331508becbca6a0735e4c1459eea2fb355b8ab/skimage/conftest.py#L4